google/cloud/bigtable/internal/metrics_test.cc

```diff
@@ -76,6 +254,534 @@ TEST(GetResponseParamsFromMetadata, EmptyHeader) {
   EXPECT_FALSE(result);
 }
 
+std::unordered_map<std::string, std::string> MakeAttributesMap(
+    opentelemetry::common::KeyValueIterable const& attributes) {
+  std::unordered_map<std::string, std::string> m;
+  attributes.ForEachKeyValue([&](opentelemetry::nostd::string_view k,
+                                 opentelemetry::common::AttributeValue v) {
+    if (opentelemetry::nostd::holds_alternative<
+            opentelemetry::nostd::string_view>(v)) {
+      m.emplace(
+          std::string{k},
+          opentelemetry::nostd::get<opentelemetry::nostd::string_view>(v));
+    }
+    return true;
+  });
+  return m;
+}
+
+void SetClusterZone(grpc::ClientContext& client_context,
+                    std::string const& cluster, std::string const& zone) {
+  google::bigtable::v2::ResponseParams expected_response_params;
+  expected_response_params.set_cluster_id(cluster);
+  expected_response_params.set_zone_id(zone);
+  RpcMetadata server_metadata;
+  server_metadata.trailers.emplace(
+      "x-goog-ext-425905942-bin", expected_response_params.SerializeAsString());
+  SetServerMetadata(client_context, server_metadata);
+}
+
+TEST(OperationLatencyTest, FirstAttemptSuccess) {
+  auto mock_histogram = std::make_unique<MockHistogram<double>>();
+  EXPECT_CALL(
+      *mock_histogram,
+      Record(A<double>(), A<opentelemetry::common::KeyValueIterable const&>(),
```

Comment:
Shall we also look into adding integration testing to verify these scenarios? Maybe as a follow-up.